### PR TITLE
fix(macos,ios): exclude background conversations from dock badge count

### PR DESCRIPTION
**Pull request by siddseethepalli**
## Summary
- Dock badge counted heartbeat/task/auto-analysis via a hardcoded source allowlist but silently included other daemon-created background conversations (watcher has `source: nil`, filing has `source: "filing"`) that the server tags with `conversationType: "background"` — those were contributing to the dock badge despite being automated.
- Plumb `conversationType` through `ConversationModel` (macOS) and `IOSConversation` + `PersistedConversation` (iOS) and widen `shouldSuppressUnreadIndicator` to check `conversationType == "background" || conversationType == "scheduled"` ahead of the existing source/title heuristics, so any automation thread the server marks is excluded regardless of its `source`.
- Add regression tests exercising filing (source="filing"), watcher (source=nil) and scheduled cases through the conversation-list response; keep a guard that a standard user conversation with an unseen assistant message still contributes to the badge.

## Original prompt
it

Unread background conversations shouldn't contribute to the notification badge count

### CODE REVIEW

**Comment by siddseethepalli** [comment]
Addressed in https://github.com/vellum-ai/vellum-assistant/pull/26456

File: clients/macos/vellum-assistant/Features/MainWindow/ConversationModel.swift [hunk 1]
```diff
@@ -97,9 +104,17 @@ struct ConversationModel: Identifiable, Hashable {
...
+    ///
+    /// Primary signal is `conversationType` — the daemon's canonical field — so any
+    /// server-created background/scheduled conversation is suppressed regardless of its
+    /// `source`. The source-based fallbacks cover locally-created conversations (before
+    /// the server round-trip sets `conversationType`) and older daemons that don't return
+    /// the field.
     var shouldSuppressUnreadIndicator: Bool {
-        isScheduleConversation || shouldReturnToBackgroundOnUnpin
+        conversationType == "background" || conversationType == "scheduled"
+            || isScheduleConversation || shouldReturnToBackgroundOnUnpin
```
[the remark this answers is not in the document]
Comment (siddseethepalli) [reply clients/macos/vellum-assistant/Features/MainWindow/ConversationModel.swift:117]: Addressed in https://github.com/vellum-ai/vellum-assistant/pull/26456
